rtf_vis_tool/src/Components/Bundle_Adj_PC.js

```diff
@@ -0,0 +1,125 @@
+/* Component that renders the Point Cloud and Camera Frustums before Bundle Adjustment. Spawned 
+once the 'SfMData' node is clicked.
+
+Author: Adi Singh
+*/
+import React, {useEffect, useState} from "react";
+
+// Third-Party Package Imports.
+import {Canvas} from "react-three-fiber"; // Used to render canvas containing 3D elements
+
+// Local Imports.
+import CoordinateGrid from './CoordinateGrid';
+import OrbitControlsComponent from './OrbitControlsComponent';
+import PointMesh from './PointMesh';
+import '../stylesheets/Bundle_Adj_PC.css';
+
+const Bundle_Adj_PC = (props) => {
+    // Variables to store the point cloud information and toggle the coordinate grid display.
+    const [pointCloudRaw, setPointCloudRaw] = useState([]);
+    const [pointCloudJSX, setPointCloudJSX] = useState([]);
+    const [showCoordGrid, setShowCoordGrid] = useState(true);
+
+    // Point size defined by radius, initialized as 0.15.
+    const [pointRadius, setPointRadius] = useState(0.15);
+    const pointSizeArr = [pointRadius]; 
+
+    // Render points3D.txt from COLMAP ba_input directory.
+    useEffect(() => {
+        // Fetch the COLMAP file from the public directory.
+        fetch('results/ba_input/points3D.txt')
+            .then(function(response){
+                return response.text();
+            })
+            .then(function(data){
```

Comment:
Consider defining function somewhere else rather than inline, with a descriptive name and it’s own comment